### PR TITLE
[Term Entry] Swift Protocols Decoder #2960

### LINKED ISSUE #2960: [Term Entry] Swift Protocols Decoder
### Reason/inspiration (optional)

We would like a new term entry for Swift [Protocols](https://www.codecademy.com/resources/docs/swift/protocols) about the Decoder protocol. The entry should go in a new file in docs/content/swift/concepts/protocols/terms/decoder/decoder.md

The entry should
    - Have an introductory paragraph about what the protocol is and does.
    - A syntax section with a pseudo code block with an explanation of it.
    - A basic example showing the protocol in use

Please refer to the [term entry templates](https://github.com/Codecademy/docs/blob/main/documentation/term-entry-template.md), [content standards](https://github.com/Codecademy/docs/blob/main/documentation/content-standards.md), and [markdown style guide](https://curriculum-documentation.codecademy.com/content-guidelines/markdown-style-guide/) when working on the PR for this issue.

### Entry Type (select all that apply)

New Term for Existing Concept Entry

### Code of Conduct

- [X] By submitting this issue, I agree to follow Codecademy Doc's [Code of Conduct](https://github.com/Codecademy/docs/blob/main/.github/CODE_OF_CONDUCT.md).

### For Maintainers

- [X] Labels added
- [x] Issue is assigned

### PR DESCRIPTION
**Pull request by marjoriekohn**
### Description

Created decoder.md for Swift > concepts > protocols > terms directory.
Closes #2960 

### Type of Change

- Adding a new entry

### Checklist

- [x] All writings are my own.
- [x] My entry follows the Codecademy Docs style guide.
- [x] My changes generate no new warnings.
- [x] I have performed a self-review of my own writing and code.
- [x] I have checked my entry and corrected any misspellings.
- [x] I have made corresponding changes to the documentation if needed.
- [x] I have confirmed my changes are not being pushed from my forked `main` branch.
- [x] I have confirmed that I'm pushing from a new branch named after the changes I'm making.
- [x] Under "Development" on the right, I have linked any issues that are relevant to this PR (write "Closes #<issue number> in the "Description" above).

### CODE REVIEW

File: content/swift/concepts/protocols/terms/decoder/decoder.md [hunk 1]
```diff
@@ -0,0 +1,71 @@
+---
+Title: 'Decoder'
+Description: 'A Swift protocol that converts external data into Swift types.'
+Subjects:
+  - 'iOS'
+  - 'Mobile Development'
+Tags:
+  - 'Swift'
```
Comment (avdhoottt) [inline content/swift/concepts/protocols/terms/decoder/decoder.md:8]:
```suggestion
```
please choose the appropriate tags from the official [documentation](https://github.com/Codecademy/docs/blob/main/documentation/tags.md). Swift is not listed in it.

File: content/swift/concepts/protocols/terms/decoder/decoder.md [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+  - 'iOS'
+  - 'Protocols'
+CatalogContent:
+  - 'learn-swift'
```
Comment (avdhoottt) [inline content/swift/concepts/protocols/terms/decoder/decoder.md:12]:
```suggestion
- 'learn-swift'
- 'paths/build-ios-apps-with-swiftui'
```
Two CatlogContent slugs should be there, first for a free course and second for a pro course. You can find the CatalogContent [here](https://github.com/Codecademy/docs/blob/main/documentation/catalog-content.md).

File: content/swift/concepts/protocols/terms/decoder/decoder.md [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+}
+
+let decoder = JSONDecoder()
+let itemInstance = try decoder.decode(SomeStruct.self, from: itemData)
+```
+
+- The `Decodable` protocol signifies that `SomeStruct` can be decoded from an external format.
+- If needed, the `init(from:)` initializer is used to provide custom decoding logic.
+- The `JSONDecoder` class decodes JSON data into a Swift type that conforms to the `Decodable` protocol.
+- JSON data gets transformed into the `SomeStruct` Swift type via the `decode(_:from:)` method.
```
Comment (avdhoottt) [inline content/swift/concepts/protocols/terms/decoder/decoder.md:35]:
```suggestion
- The `Decodable` protocol signifies that `SomeStruct` can be decoded from an external format.
- If needed, the `init(from:)` initializer is used to provide custom decoding logic.
- The `JSONDecoder` class decodes JSON data into a Swift type that conforms to the `Decodable` protocol.
- JSON data gets transformed into the `SomeStruct` Swift type via the `decode(_:from:)` method.
```
Explain the itemData parameter too.

File: content/swift/concepts/protocols/terms/decoder/decoder.md [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+
+## Example
+
+In the following example, a JSON representation of a book is translated into a `Book` instance which conforms to the `Decodable` protocol.
```
Comment (avdhoottt) [inline content/swift/concepts/protocols/terms/decoder/decoder.md:39]:
```suggestion
In the following example, a JSON representation of a book is translated into a `Book` instance.
```
The sentence is repeated many times which is not good for readability.

**Review by avdhoottt — CHANGES_REQUESTED** [review]
@marjoriekohn I've suggested some changes, please make them.

File: content/swift/concepts/protocols/terms/decoder/decoder.md [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+- The `init(from:)` initializer is optional and provides a way to implement custom decoding logic.
+- Instantiate a decoder class, such as `JSONDecoder`, to convert the encoded data into the desired Swift type.
+- `someData` is a `Data` object containing the data to decode.
+- The `decode(_:from:)` method transforms the data into an instance of `SomeStruct`.
+
+## Example
+
+In the following example, a JSON object is decoded into a `Book` struct.
+
+```pseudo
```
Comment (avdhoottt) [inline content/swift/concepts/protocols/terms/decoder/decoder.md:41]:
```suggestion
```swift
```
The example should be in a code block.

**Review by avdhoottt — APPROVED** [review]
PR is ready for the second review.

File: content/swift/concepts/protocols/terms/decoder/decoder.md [hunk 6]
```diff
@@ -0,0 +1,68 @@
+---
+Title: 'Decoder'
+Description: 'A Swift protocol that converts external data into Swift types.'
```
Comment (ishg-153) [inline content/swift/concepts/protocols/terms/decoder/decoder.md:3]:
```suggestion
Description: 'Converts external data into Swift types.'
```

**Review by ishg-153 — CHANGES_REQUESTED** [review]
@marjoriekohn There's one small change you need to make.

**Review by ishg-153 — APPROVED** [review]
@marjoriekohn Great! Ready to merge.

**Pull request merged**